kairon/shared/utils.py

```diff
@@ -132,6 +132,21 @@ def check_empty_string(value: str):
         else:
             return False
 
+    @staticmethod
+    def check_data_type(data: Any, metadata: Dict):
+        if metadata and isinstance(data, dict):
+            column_name = metadata["column_name"]
+            data_type = metadata["data_type"]
+            if column_name in data:
```

Comment:
column_name is not in data, then this returns None and creating embedding will fail